pytorch_lightning/callbacks/progress/rich_progress.py

```diff
@@ -346,32 +344,35 @@ def on_train_epoch_start(self, trainer, pl_module):
         self.refresh()
 
     def on_validation_epoch_start(self, trainer, pl_module):
-        if self.total_val_batches > 0:
-            total_val_batches = self.total_val_batches
-            if self.total_train_batches != float("inf") and hasattr(trainer, "val_check_batch"):
-                # val can be checked multiple times per epoch
-                val_checks_per_epoch = self.total_train_batches // trainer.val_check_batch
-                total_val_batches = self.total_val_batches * val_checks_per_epoch
-            self.val_progress_bar_id = self._add_task(total_val_batches, self.validation_description, visible=False)
-            self.refresh()
+        if trainer.sanity_checking:
+            self.val_sanity_progress_bar_id = self._add_task(self.total_val_batches, self.sanity_check_description)
+        else:
+            self.val_progress_bar_id = self._add_task(
+                self.total_val_batches, self.validation_description, visible=False
+            )
+
+        self.refresh()
 
     def _add_task(self, total_batches: int, description: str, visible: bool = True) -> Optional[int]:
         if self.progress is not None:
             return self.progress.add_task(
                 f"[{self.theme.description}]{description}", total=total_batches, visible=visible
             )
 
-    def _update(self, progress_bar_id: int, current: int, total: int, visible: bool = True) -> None:
+    def _update(self, progress_bar_id: int, current: int, total: Union[int, float], visible: bool = True) -> None:
         if self.progress is not None and self._should_update(current, total):
```

Comment:
maybe reverse logic, also easier if you need to put breakpoint to return...
```py
        if self.progress is None or not self._should_update(current, total):
            return
        leftover = current % self.refresh_rate
        advance = leftover if (current == total and leftover != 0) else self.refresh_rate
        self.progress.update(progress_bar_id, advance=advance, visible=visible)
        self.refresh()
```

Reply:
I always wonder, is this a better code pattern from what's already there?